Public/Src/Engine/Dll/EngineSchedule.cs

```diff
@@ -771,21 +772,43 @@ private bool ShouldSerializeOptimizationDataStructurePostExecution()
                 outputDirectories = scheduler.PipGraph.AllDirectoriesContainingOutputs().Select(d => d.ToString(scheduler.Context.PathTable)).ToList();
             }
 
+            var scrubber = new DirectoryScrubber(
+                cancellationToken: scheduler.Context.CancellationToken,
+                loggingContext: loggingContext,
+                loggingConfiguration: configuration.Logging,
+                maxDegreeParallelism: Environment.ProcessorCount,
+                tempDirectoryCleaner: tempCleaner);
+
+            var sharedOpaqueSidebandDirectory = configuration.Layout.SharedOpaqueSidebandDirectory.ToString(scheduler.Context.PathTable);
+            var sharedOpaqueSidebandFiles = SharedOpaqueOutputLogger.FindAllProcessPipSidebandFiles(sharedOpaqueSidebandDirectory);
+            var distinctRecordedWrites = sharedOpaqueSidebandFiles
+                .AsParallel()
+                .WithDegreeOfParallelism(Environment.ProcessorCount)
+                .WithCancellation(scheduler.Context.CancellationToken)
+                .SelectMany(SharedOpaqueOutputLogger.ReadRecordedPathsFromSidebandFile)

```

Comment:
>.SelectMany(SharedOpaqueOutputLogger.ReadRecordedPathsFromSidebandFile) [](start = 16, length = 71)

Don't you need a catch block here for the corrupted file case? (and I think you could limit it to a particular file being corrupted, and let the other potential sideband files to be honored)

Reply:
`SharedOpaqueOutputLogger.ReadRecordedPathsFromSidebandFile` shouldn't throw unless something really really bad happens (in which case we should probably know about it and not swallow it)

Reply:
ok, I guess I was missing what happens when the file is corrupted then (which I thought it would be about ignoring the corrupted file)

---
In reply to: [334722307](https://github.com/microsoft/BuildXL/pull/1029#discussion_r334722307) [](ancestors = 334722307)